components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/Utils.kt

```diff
@@ -0,0 +1,23 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.fxa
+
+/**
+ * Runs a provided lambda, and if that throws non-panic FxA exception, runs the second lambda.
+ */
+fun <T> maybeExceptional(block: () -> T, onFailure: (e: FxaException) -> T): T {
```

Comment:
nice!

nit: I'd find `handleError` or `withErrorHandling` clearer here.

Reply:
ah, yeah, that's clearer. naming things is hard :)